executor/kava/executor.go

```diff
@@ -98,22 +93,28 @@ func (executor *Executor) GetBlockAndTxs(height int64) (*common.BlockAndTxLogs,
 		for _, msg := range msgs {
 			switch realMsg := msg.(type) {
 			case bep3.MsgCreateAtomicSwap:
-				if !realMsg.CrossChain {
-					continue
-				}
-
 				if len(realMsg.Amount) != 1 {
 					continue
 				}
 
 				signer := msg.GetSigners()[0]
 				randomNumberHash := hex.EncodeToString(realMsg.RandomNumberHash)
 
-				// Regex matches (key, value) for swap ID in logs
-				m := regexp.MustCompile(`{"key":"atomic_swap_id","value":"[a-zA-Z0-9]{64}"}`)
-				res := m.FindString(txResult.Log)
-				// Trim result prefix and suffix to isolate swap ID
-				swapID := res[33:97]
+				// Parse swap ID from create atomic swap event
+				var swapID string
+				for _, event := range txResult.Events {
+					if event.GetType() == "create_atomic_swap" {
+						for _, attribute := range event.GetAttributes() {
+							if string(attribute.GetKey()) == "atomic_swap_id" {
+								swapID = string(attribute.GetValue())
+							}
+						}
+					}
+				}
+				if len(strings.TrimSpace(swapID)) == 0 {
+					util.Logger.Errorf("err='atomic_swap_id' event attribute not found")
```

Comment:
do we need to add `err=` as a prefix

Reply:
Removed the prefix in https://github.com/binance-chain/bep3-deputy/pull/2/commits/371a26166b1c28d14679c0c1f23a6ba9b871cdda